src/test/java/org/candlepin/model/EnvironmentTest.java

```diff
@@ -42,4 +47,52 @@ public void testGetOwnerKeyWithNoOwner() {
         assertNull(environment.getOwnerKey());
     }
 
+    @Test
+    public void testSetContentPrefix() {
+        Environment environment = new Environment();
+        Date before = minusTwoSeconds();
+        environment.setLastContentUpdate(before);
+
+        assertNull(environment.getContentPrefix());
+
+        environment.setContentPrefix("newPrefix");
+        Date after = environment.getLastContentUpdate();
+
+        assertEquals("newPrefix", environment.getContentPrefix());
+        assertTrue(after.after(before));
+    }
+
+    @Test
+    public void testSetLastContentUpdate() {
+        Environment environment = new Environment();
+        Date date = minusTwoSeconds();
+
+        assertThrows(IllegalArgumentException.class, () -> environment.setLastContentUpdate(null));
+
+        environment.setLastContentUpdate(date);
+        assertEquals(date, environment.getLastContentUpdate());
+    }
+
+    @Test
+    public void testGetLastContentUpdate() {
```

Comment:
We should add more testing around this method to verify the various conditions we've defined in the comments and design.

This does get a bit spicy with the constructors calling syncLastContentUpdate. I'm somewhat torn on if that should be a thing since it's a tad extraneous and could lead to incorrect results if Hibernate gets its order of operations wrong. But if it gets in the way of us verifying output, maybe it should be cut. I'll leave that decision to you.

Reply:
Based on discussions in slack, this comment can be disregarded.

For those not part of the discussion: if we don't set the lastContentUpdate field during construction (either directly setting it or calling sync/setLastContentUpdate), it leaves the entity in an invalid state with a null LCU which will get rejected by the backing DB schema. So while testing does lose some granularity, I would rather not introduce the potential for invalid or undefined states for the sake of easing our unit tests.